datadogfirehosenozzle/datadog_firehose_nozzle.go

```diff
@@ -30,8 +30,8 @@ type AuthTokenFetcher interface {
 func NewDatadogFirehoseNozzle(config *nozzleconfig.NozzleConfig, tokenFetcher AuthTokenFetcher) *DatadogFirehoseNozzle {
 	return &DatadogFirehoseNozzle{
 		config:           config,
-		errs:             make(chan error),
-		messages:         make(chan *events.Envelope),
+		errs:             make(<-chan error),
+		messages:         make(<-chan *events.Envelope),
```

Comment:
I think these lines are made unnecessary by line 69, below.  These channels should be left nil until they are created by a call to consumer.